genvm/vm_test.go

```diff
@@ -431,6 +431,9 @@ func (t *tester) rewards(all ...reward) []types.CoinbaseReward {
 		rat := new(big.Rat).SetFloat64(rew.share)
 		rst = append(rst, types.CoinbaseReward{
 			Coinbase: t.accounts[rew.address].getAddress(),
+			// there can be many rewards per coinbase in the same layer,
+			// but each must have a distinct smesherID.
+			SmesherID: types.RandomNodeID(),
```

Comment:
types.NodeID{byte(i)} . those types.RandomX is a mistake and should not be used at all

Reply:
`types.BytesToNodeID(bytes)` is a helper that can also be used to convert `[]byte` to `NodeID`.

Why do you not like `types.RandomX`? Especially for tests they are quite useful in my opinion.

Reply:
they are not deterministic, i prefer to use randomness generated from constant seed or use deterministic values that, if that is simple enough.